x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryFolder.java

```diff
@@ -143,9 +143,17 @@ protected PhysicalPlan rule(ProjectExec project) {
                 AttributeMap.Builder<Expression> aliases = AttributeMap.<Expression>builder().putAll(queryC.aliases());
                 AttributeMap.Builder<Pipe> processors = AttributeMap.<Pipe>builder().putAll(queryC.scalarFunctions());
 
+                // recreate the query container's fields such that they appear in order of the projection and with non-projected fields
+                // last. This is mostly needed for PIVOT queries where we have to fold projections on aggregations because they cannot be
+                // optimized away. Most (all) other queries usually prune nested projections in earlier steps.
+                List<QueryContainer.FieldInfo> fields = new ArrayList<>(queryC.fields().size());
+                List<QueryContainer.FieldInfo> nonProjectedFields = new ArrayList<>(queryC.fields());
```

Comment:
I find `nonProjected` confusing as it might imply a field that was not specified in the projection (maybe because of an implicit declaration) - it's better to describe what it does as oppose to what it does NOT do.
How about `extraFields`,  `hiddenFields`  or `infraFields` and update the comment re: non-projected fields.